src/main/java/uk/gov/hmcts/darts/audit/api/AuditApi.java

```diff
@@ -5,7 +5,20 @@
 
 public interface AuditApi {
 
-    void record(AuditActivity activity, UserAccountEntity userAccountEntity, CourtCaseEntity courtCase);
+
+    default void record(AuditActivity activity, UserAccountEntity userAccountEntity) {
+        record(activity, userAccountEntity, null, null);
```

Comment:
If we can knowingly accept a null maybe Optional would be better so implementations can explicitly handle a null value if they need to

Reply:
Relates to https://github.com/hmcts/darts-api/pull/2094

Reply:
I'm happy for this to be updated once #2094 is merged

Reply:
The designers of Optional explicitly warn against using it as method parameters (e.g. https://stackoverflow.com/a/26328555):
"...You should almost never use it as a field of something or a method parameter."